### PR TITLE
🌱 refactor AWSMachine to use scope interfaces

### LINKED ISSUE #1852: Use new scope interfaces in AWSMachine reconciliation
/kind feature

@richardcase did a bunch of great work setting up interfaces to make working with AWSCluster and AWSManagedControlPlane more simple.

For provisioning EC2 instances to work for both the managed and unmanaged controlplanes, we need to consume these interfaces in the AWSMachine reconciliation.

This means:
- Need to use cloud.ClusterScoper, ec2.Scope, and elb.Scope instead of ClusterScope where appropriate
- Need to audit any calls to scope.AWSCluster.Foo and determine if we need to either a. add more methods to the ClusterScoper interface, or implement some kind of type switching. For example, ClusterScoper currently doesn't have any ImageLookup() methods, and the AWSMachine reconciler checks AWSCluster.Spec.ImageLookupOrg.

### PR DESCRIPTION
**Pull request by rudoi**
**What this PR does / why we need it**: Refactors AWSMachine reconciliation to utilize the new scope interfaces. This will make onboarding the AWSManagedControlPlane stuff more simple.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1852

### CODE REVIEW

File: controllers/awsmachine_controller.go [hunk 1]
```diff
@@ -406,7 +403,7 @@ func (r *AWSMachineReconciler) findInstance(scope *scope.MachineScope, ec2svc se
 	return instance, nil
 }
 
-func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *scope.MachineScope, clusterScope *scope.ClusterScope) (ctrl.Result, error) {
+func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper, ec2Scope scope.EC2Scope, elbScope scope.ELBScope) (ctrl.Result, error) {
```
Comment (rudoi) [inline controllers/awsmachine_controller.go:406]: I don't know if I was supposed to do it this way...

**Comment by rudoi** [comment]
added WIP while I address changes made in #1856

**Comment by rudoi** [comment]
unwipped, this is good to go i think

cc @richardcase @randomvariable

**Comment by randomvariable** [comment]
/approve
/assign @richardcase
for lgtm

File: controllers/awsmachine_controller.go (hunk 1 above)
Comment (randomvariable) [reply controllers/awsmachine_controller.go:408]: Ideally, switch this over to `reconcileNormalInput(input)`, but can do as follow up.

File: pkg/cloud/scope/machine.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func NewMachineScope(params MachineScopeParams) (*MachineScope, error) {
 	if params.AWSMachine == nil {
 		return nil, errors.New("aws machine is required when creating a MachineScope")
 	}
-	if params.AWSCluster == nil {
+	if params.InfraCluster == nil {
```
Comment (richardcase) [inline pkg/cloud/scope/machine.go:64]: Wondering if the error message needs updating? Probably not

**Pull request merged**